modules/caddypki/acmeserver/acmeserver.go

```diff
@@ -142,31 +165,30 @@ func (ash *Handler) Provision(ctx caddy.Context) error {
 		DB: database,
 	}
 
-	auth, err := ca.NewAuthority(authorityConfig)
+	ash.acmeAuth, err = ca.NewAuthority(authorityConfig)
 	if err != nil {
 		return err
 	}
 
-	var acmeDB acme.DB
-	if authorityConfig.DB != nil {
-		acmeDB, err = acmeNoSQL.New(auth.GetDatabase().(nosql.DB))
-		if err != nil {
-			return fmt.Errorf("configuring ACME DB: %v", err)
-		}
+	ash.acmeDB, err = acmeNoSQL.New(ash.acmeAuth.GetDatabase().(nosql.DB))
+	if err != nil {
+		return fmt.Errorf("configuring ACME DB: %v", err)
```

Comment:
@mholt has some opinions about wrapping errors so :man_shrugging: I'll let him tell me if he wants me to do this or not. But noted.

Reply:
It just means you have to be OK with error values being part of the exported API. Since `Provision()` is called by the caddy core, there's a good chance it won't know what to type-assert the error as, and thus exporting it here won't hurt but also won't help IMO.

Great tip though @maraino -- thanks for pointing that out!